### ISSUE TITLE
Don't use placeholder text only for form input labels

### ISSUE THREAD

**Issue by HaSistrunk** [MutualAidNYC/hsdirectory#4]
## Is your feature request related to a problem? Please describe.
The form search inputs use placeholder text as labels. This is [generally bad practice for accessibility and usability](https://www.nngroup.com/articles/form-design-placeholders/), and doesn't work well on small screens because the text is obscured.

## Describe the solution you'd like
Move the placeholder text out of the input and make it a responsive visible label above the input

## Describe alternatives you've considered
We can also keep the placeholder text and add an additional label, if desired. If necessary, the labels can be visually hidden but made accessible to screen readers.

## Additional context
Example:
<img width="694" height="78" alt="Image" src="https://github.com/user-attachments/assets/71f03016-31e9-4b09-8418-53225b4ad856" />
